corehq/apps/geospatial/views.py

```diff
@@ -435,3 +439,110 @@ def get_users_with_gps(request, domain):
     ]
 
     return json_response({'user_data': user_data})
+
+
+@method_decorator(toggles.GEOSPATIAL.required_decorator(), name="dispatch")
+class CasesReassignmentView(BaseDomainView):
+    urlname = "reassign_cases"
+    MAX_REASSIGNMENT_REQUEST_CASES = 100
+    SYNC_CASES_UPDATE_THRESHOLD = 1000
+    ASYNC_CASES_LIMIT = 5000
+
+    def post(self, request, domain, *args, **kwargs):
+        try:
+            request_data = json.loads(request.body)
+        except json.decoder.JSONDecodeError:
+            return HttpResponseBadRequest(
+                _('POST Body must be a valid json')
+            )
+
+        case_id_to_owner_id = request_data.get('case_id_to_owner_id', {})
+        include_related_cases = request_data.get('include_related_cases')
+        if len(case_id_to_owner_id) > self.MAX_REASSIGNMENT_REQUEST_CASES:
+            return HttpResponseBadRequest(
+                _("Maximum number of cases that can be reassigned is {limit}").format(
+                    limit=self.MAX_REASSIGNMENT_REQUEST_CASES
+                )
+            )
+
+        if include_related_cases:
+            request_cases_id = list(case_id_to_owner_id.keys())
+            parent_to_child_cases_id = self.get_child_cases(domain, request_cases_id)
+            for parent_case_id, child_cases_ids in parent_to_child_cases_id.items():
+                for child_case_id in child_cases_ids:
+                    self._add_related_case(case_id_to_owner_id, parent_case_id, child_case_id)
+
+            child_to_parent_case_id = self.get_parent_cases(domain, request_cases_id)
+            for child_case_id, parent_case_id in child_to_parent_case_id.items():
+                self._add_related_case(case_id_to_owner_id, child_case_id, parent_case_id)
+
+        if len(case_id_to_owner_id) > self.ASYNC_CASES_LIMIT:
+            return HttpResponseBadRequest(
+                _("Maximum limit for cases to be reassigned including related cases exceeded."
+                  " Please select a lower value to update at time or consider deselecting 'include related cases'."
```

Comment:
Nit: Update to say
```
Case reassignment limit exceeded. Please select fewer cases to update or consider deselecting 'include related cases'
```

Two things:
1. Shorter messages are easier to read
2. The "including related cases" part in the first sentence can be deducted from the second where the suggestion to deselect "include related cases" is made. 

Feel free to disagree and/or ignore this comment :)

Reply:
Addressed in fcdba68f6a55349f35693160ecc756ed2947fdbd